### PR TITLE
Add different player boards for different amount of players

### PR DESCRIPTION
**Pull request by Dammic**
Also, added curtain effect on starting new game

### CODE REVIEW

File: utils/RoomsManager.js [hunk 1]
```diff
@@ -473,6 +489,23 @@ class RoomsManager {
...
+        let boardType
+        if (playersCount <= 6) {
+            boardType = PlayerBoards.SmallBoard
+        } else if (playersCount <= 8) {
+            boardType = PlayerBoards.MediumBoard
+        } else if (playersCount <= 10) {
+            boardType = PlayerBoards.LargeBoard
+        }
+        this.rooms_props[roomName].boardType = boardType
+    }
```
Comment (Terverok) [inline utils/RoomsManager.js]: Please add test to this function :smiley:

File: src/features/PlayerBoard/CurtainWrapper/CurtainWrapper.scss [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        .curtain__panel {
+            background-color: #79040A; 
+            background-image: url("../static/curtain.png");
+        }
+    }
+
+    &.curtain__panel--liberal {
+        .curtain__panel {
+            background-color: #063F56; 
+            background-image: url("../static/curtain.png");
```
Comment (Terverok) [inline src/features/PlayerBoard/CurtainWrapper/CurtainWrapper.scss]: Maybe it's better to move that `background-image` higher to more general class `.curtain__panel`? Now it repeats in both liberal and fascist curtain classes.

File: events/PhaseSocketEvents.js [hunk 3]
```diff
@@ -8,6 +8,7 @@ const PhaseSocketEvents = (io, RoomsManager) => {
     const phaseSocketEvents = {
         startGame: (socket) => {
             RoomsManager.startGame(socket.currentRoom)
+            RoomsManager.setPlayerboardType(socket.currentRoom)
```
Comment (Terverok) [inline events/PhaseSocketEvents.js]: Is there a case when we change Playerboad type during the game? Maybe you should call this function inside `RoomManager.startGame()`?

File: src/features/PlayerBoard/CurtainWrapper/CurtainWrapper.scss [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+            transform: translateX(150%);
+        }
+    }
+
+    .curtain__prize {
+        position: absolute;
+        z-index: 1;
+        width: 100%;
+        height: 100%;
+    }
```
Comment (Dammic) [inline src/features/PlayerBoard/CurtainWrapper/CurtainWrapper.scss]: unused css style

File: src/features/PlayerBoard/CurtainWrapper/CurtainWrapper.scss (hunk 2 above)
Comment (Dammic) [reply src/features/PlayerBoard/CurtainWrapper/CurtainWrapper.scss]: Good point, moving it right now

File: utils/RoomsManager.js (hunk 1 above)
Comment (Dammic) [reply utils/RoomsManager.js]: Tests added

File: events/PhaseSocketEvents.js (hunk 3 above)
Comment (Dammic) [reply events/PhaseSocketEvents.js]: That could be wise

File: utils/__tests__/RoomsManager.spec.js [hunk 5]
```diff
@@ -165,4 +167,127 @@ describe('RoomsManager', () => {
...
+                player9: {},
+                player10: {},
+                player11: {},
+            }
+            const initialRoomProps = cloneDeep(RoomsManager.rooms_props['testRoom'])
+            initialRoomProps.boardType = undefined
+            RoomsManager.setPlayerboardType('testRoom')
+            expect(initialRoomProps).toEqual(RoomsManager.rooms_props['testRoom'])
+        })
+    })
```
Comment (Terverok) [inline utils/__tests__/RoomsManager.spec.js]: You couldn't do more code duplication? :full_moon_with_face: Wouldn't it be better to create one helper function that creates given amount of fake players objects and initial boardType to given as second argument?

File: utils/__tests__/RoomsManager.spec.js [hunk 6]
```diff
@@ -169,117 +169,36 @@ describe('RoomsManager', () => {
...
-            }
-            const initialRoomProps = cloneDeep(RoomsManager.rooms_props['testRoom'])
-            initialRoomProps.boardType = undefined
-            RoomsManager.setPlayerboardType('testRoom')
-            expect(initialRoomProps).toEqual(RoomsManager.rooms_props['testRoom'])
+        const generatePlayersDict = (amount) => {
+            let playersKeys = {}
+            times(amount, (index) => {
+                playersKeys[`player${index + 1}`] = {}
+            })
```
Comment (Terverok) [inline utils/__tests__/RoomsManager.spec.js]: Można zrobić to za pomocą `reduce'a`, np. tak:
`return reduce(times(amount), (obj, index) => { obj[`player${index + 1}`] = {}; return obj; }, {})

File: utils/__tests__/RoomsManager.spec.js [hunk 7]
```diff
@@ -169,117 +169,36 @@ describe('RoomsManager', () => {
...
+            11: undefined,
+        }
+        forEach(expectedResults, (expectedBoardSize, playersCount) => {
+            test(`Should set ${expectedBoardSize} for ${playersCount} players`, () => {
+                RoomsManager.rooms_props['testRoom'].playersDict = generatePlayersDict(playersCount)
+                const initialRoomProps = cloneDeep(RoomsManager.rooms_props['testRoom'])
+                initialRoomProps.boardType = expectedBoardSize
+                RoomsManager.setPlayerboardType('testRoom')
+                expect(initialRoomProps).toEqual(RoomsManager.rooms_props['testRoom'])
+            })
```
Comment (Terverok) [inline utils/__tests__/RoomsManager.spec.js]: Very nice :+1: :+1: :)

**Pull request merged**